ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_creation.py

```diff
@@ -27,6 +28,18 @@ def _random_cp_data(draw):
     return shape, rank, dtype[0], full, orthogonal, seed, normalise_factors
 
 
+@st.composite
+def _random_tr_data(draw):
+    shape = draw(
+        st.lists(helpers.ints(min_value=1, max_value=5), min_size=2, max_size=4)
+    )
+    rank = min(shape)
+    dtype = draw(st.sampled_from(["float32", "float64"]))
```

Comment:
Why can't we just specify `float`?

Reply:
because `torch.matmul` and `paddle.matmul` doesn't support `float16`, so they make the tests fails

Reply:
The torch backend implementation of `torch.matmul` already contains `float16` in the `unsupported_dtypes` decorator. In your [PR](https://github.com/unifyai/ivy/pull/22700) for removing manual casting from our backend implementations and adding the `unsupported_dtypes` decorator if a data-type is not supported, i guess you will also remove the casting inside `paddle.matmul` and add the `unsupported_dtypes` decorator right? If the only reason for failing test cases is the unsupported dtypes issue, let's change this `valid` and merge it :)

Reply:
understood

Reply:
Actually there are some changes that are not covered by the [PR](https://github.com/unifyai/ivy/pull/22700). 
I made them and now the tests pass except for paddle which will be passing when the PR is merged.